pkg/controller/podgc/gc_controller_test.go

```diff
@@ -51,41 +52,32 @@ func NewFromClient(kubeClient clientset.Interface, terminatedPodThreshold int) (
 	return controller, podInformer, nodeInformer
 }
 
-func compareStringSetToList(set sets.String, list []string) bool {
-	for _, item := range list {
-		if !set.Has(item) {
-			return false
-		}
-	}
-	if len(list) != len(set) {
-		return false
-	}
-	return true
-}
-
-func getDeletedPodNames(client *fake.Clientset) []string {
-	deletedPodNames := make([]string, 0)
-	for _, action := range client.Actions() {
-		if action.GetVerb() == "delete" && action.GetResource().Resource == "pods" {
-			deleteAction := action.(clienttesting.DeleteAction)
-			deletedPodNames = append(deletedPodNames, deleteAction.GetName())
-		}
-	}
-	return deletedPodNames
-}
-
 func TestGCTerminated(t *testing.T) {
 	type nameToPhase struct {
 		name  string
 		phase v1.PodPhase
 	}
 
 	testCases := []struct {
-		name            string
-		pods            []nameToPhase
-		threshold       int
-		deletedPodNames sets.String
+		name                          string
+		pods                          []nameToPhase
+		threshold                     int
+		deletedPodNames               sets.String
```

Comment:
Nit: would be nice if the taint_manager_test.go is also checking deleted and patched pod names rather then just verifying the action type.

Reply:
A similar testing approach is used in gc_controller_test.go. 

This seems like a pre-existing limitation in the test, which may require an extension of the fake kube client to interpret patches. thus I suggest an independent Issue to improve the tests.